python/tests/tsutil.py

```diff
@@ -1282,3 +1567,47 @@ def genealogical_nearest_neighbours(ts, focal, reference_sets):
     L[L == 0] = 1
     A /= L.reshape((len(focal), 1))
     return A
+
+
+def assert_table_collections_equal(t1, t2, ignore_provenance=False):
+    """
+    Checks for table collection equality, but step-by-step,
+    so it's easy to see what's different.
+    """
+    if ignore_provenance:
+        t1.provenances.clear()
+        t2.provenances.clear()
+    assert_tables_equal(t1.populations, t2.populations, "populations")
+    assert_tables_equal(t1.individuals, t2.individuals, "individuals")
+    assert_tables_equal(t1.nodes, t2.nodes, "nodes")
+    assert_tables_equal(t1.edges, t2.edges, "edges")
+    assert_tables_equal(t1.sites, t2.sites, "sites")
+    assert_tables_equal(t1.mutations, t2.mutations, "mutations")
+    assert_tables_equal(t1.migrations, t2.migrations, "migrations")
+    assert_tables_equal(t1.provenances, t2.provenances, "provenances")
+    assert t1.metadata_schema == t2.metadata_schema
+    assert t1.metadata == t2.metadata
+    assert t1.metadata_bytes == t2.metadata_bytes
+    assert t1.sequence_length == t2.sequence_length
+    assert t1 == t2
+
+
+def assert_tables_equal(t1, t2, label=""):
+    if hasattr(t1, "metadata_schema"):
+        if t1.metadata_schema != t2.metadata_schema:
+            print(f"{label} :::::::::: t1 ::::::::::::")
```

Comment:
I don't think we should use prints here for a few reasons: (a) when we promote this to a real method (#1076) the message should be part of the exception that gets raised; (b) it's quite easy to miss prints in pytest anyway, and I think a lot of the people we're trying to help would miss the messages.

What about:
```python
if t1.metadata_schema != t2.metadata_schema:
    msg = (
         f"{label} :::::::::: t1 ::::::::::::\n{t1.metadata_schema}"
         f"{label} :::::::::: t2 ::::::::::::\n{t1.metadata_schema}"
   ) 
   raise AssertionError(msg)
```